braidpool-primitives/src/braid/mod.rs

```diff
@@ -23,19 +17,14 @@ pub enum AddBeadStatus {
 
 // Type Aliases
 type NumberOfBeadsUnorphaned = usize;
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug)]
 
 pub struct Braid {
     pub(crate) beads: HashSet<BeadHash>,
```

Comment:
Why have `HashSet<BeadHash>` instead of `HashSet<Bead>`?

Essentially we've created a layer of indirection where we have to get the BeadHash and then look it up in `loaded_beads_in_memory`.

What's going to go wrong in Rust if we just use the Bead object directly?

This would mean that the same objects are in both `beads`, `tips`, `cohorts`, `orphan_beads` and `genesis_beads`. In C++ I'd just use references or pointers but Rust is annoying about ownership...

Can you make an argument for using BeadHash here instead of the Bead object itself?

Can we create some kind of lazy loading, where if we try to access a bead that is not in memory it will trigger a db call to load them?